simulator-core/src/main/java/com/evbox/everon/ocpp/simulator/station/support/SecurityUtils.java

```diff
@@ -0,0 +1,77 @@
+package com.evbox.everon.ocpp.simulator.station.support;
+
+import org.bouncycastle.asn1.DERBMPString;
+import org.bouncycastle.asn1.pkcs.PKCSObjectIdentifiers;
+import org.bouncycastle.cert.jcajce.JcaX509ExtensionUtils;
+import org.bouncycastle.crypto.engines.DESedeEngine;
+import org.bouncycastle.crypto.engines.RC2Engine;
+import org.bouncycastle.crypto.modes.CBCBlockCipher;
+import org.bouncycastle.pkcs.*;
+import org.bouncycastle.pkcs.bc.BcPKCS12MacCalculatorBuilder;
+import org.bouncycastle.pkcs.bc.BcPKCS12PBEOutputEncryptorBuilder;
+import org.bouncycastle.pkcs.jcajce.JcaPKCS12SafeBagBuilder;
+
+import java.io.IOException;
+import java.security.NoSuchAlgorithmException;
+import java.security.PrivateKey;
+import java.security.PublicKey;
+import java.security.cert.X509Certificate;
+import java.util.Collections;
+import java.util.List;
+
+public final class SecurityUtils {
+
+    private SecurityUtils() {
+        // NOP
+    }
+
+    /**
+     * Generates a PCKS12 trust store containing the specified certificates.
+     *
+     * @param certificate   station certificate
+     * @param chain         certificates chain without leaf certificate
+     * @param publicKey     station public key
+     * @param privateKey    station private key
+     * @return              pkcs12 trust store with arguments passed above
+     * @throws NoSuchAlgorithmException
+     * @throws IOException
+     * @throws PKCSException
+     */
+    public static PKCS12PfxPdu generateKeyStore(X509Certificate certificate, List<X509Certificate> chain, PublicKey publicKey, PrivateKey privateKey) throws NoSuchAlgorithmException, IOException, PKCSException {
+        char[] emptyPassword = new char[0];
+
+        JcaX509ExtensionUtils extUtils = new JcaX509ExtensionUtils();
+
+        PKCS12SafeBagBuilder stationCertBagBuilder = new JcaPKCS12SafeBagBuilder(certificate);
+        stationCertBagBuilder.addBagAttribute(PKCSObjectIdentifiers.pkcs_9_at_friendlyName, new DERBMPString("Station's Key"));
+        stationCertBagBuilder.addBagAttribute(PKCSObjectIdentifiers.pkcs_9_at_localKeyId, extUtils.createSubjectKeyIdentifier(publicKey));
+
+        if (chain == null) {
+            chain = Collections.emptyList();
+        }
+
+        PKCS12SafeBag[] certs = new PKCS12SafeBag[chain.size() + 1];
```

Comment:
Why not just add to a List + in the loop below and avoid dealing with indexes? The loop also can be replaced by forEach

Reply:
Because the `addEncryptedData` method of `PKCS12PfxPduBuilder` accepts an array and not a list. I can use a list and the convert to array, but it would be slower i think

Reply:
In one-time operation it doesn't really matter, TLS handshake alone takes much longer time